std-bits/table/src/main/java/org/enso/table/data/table/join/ScanJoin.java

```diff
@@ -0,0 +1,41 @@
+package org.enso.table.data.table.join;
+
+import org.enso.table.data.table.Table;
+import org.graalvm.collections.Pair;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ScanJoin implements JoinStrategy {
+  @Override
+  public JoinResult join(Table left, Table right, List<JoinCondition> conditions) {
+    List<Pair<Integer, Integer>> matches = new ArrayList<>();
+    int ls = left.rowCount();
+    int rs = right.rowCount();
+    for (int l = 0; l < ls; ++l) {
+      for (int r = 0; r < rs; ++r) {
```

Comment:
We could do something where we get the set of `leftValues` outside the `r` loop not sure would be vastly quicker but might be useful.

Reply:
As this is a naive implementation anyway, I'd skip that optimization. It will be surpassed by proper hashmap in a moment anyway.